crates/flycheck/src/lib.rs

```diff
@@ -375,12 +383,34 @@ impl FlycheckActor {
                     }
                 }
 
-                (cmd, args)
+                if args.contains(&SAVED_FILE_PLACEHOLDER.to_owned()) {
+                    // If the custom command has a $saved_file placeholder, and
+                    // we're saving a file, replace the placeholder in the arguments.
+                    if let Some(saved_file) = saved_file {
+                        let args = args
+                            .iter()
+                            .map(|arg| {
+                                if arg == SAVED_FILE_PLACEHOLDER {
+                                    saved_file.to_string()
+                                } else {
+                                    arg.clone()
+                                }
+                            })
+                            .collect();
+                        (cmd, args)
+                    } else {
+                        // The custom command has a $saved_file placeholder,
+                        // but we had an IDE event that wasn't a file save. Do nothing.
+                        return None;
```

Comment:
This means there's going to be no initial check when loading the project, but I don't see any way around it without configuring two check commands.

I guess this is not a problem for you, but Cargo users trying this might still be surprised (ref. https://github.com/rust-lang/rust-analyzer/issues/12882).

Reply:
I suppose a workaround would be https://github.com/rust-lang/rust-analyzer/issues/15892, but yes, in this PR? it might confuse Cargo users. 

(That being said, this is very much not a feature for Cargo, but I think that #15892 _could_ make it a feature for Cargo users and resolve #12882 in the process).

Reply:
An example this should work for is `rust-lang/rust`'s cargo wrapper, `./x.py check`.

I reckon: If the command has $saved_file in it, return None. If it doesn't, return Some, as it does not require substitution. That brings back the current behaviour until you start using $saved_file.